src/DotLiquid/Tags/Html/TableRow.cs

```diff
@@ -63,20 +63,20 @@ public override void Render(Context context, TextWriter result)
 
             if (_attributes.ContainsKey("offset"))
             {
-                int offset = Convert.ToInt32(_attributes["offset"]);
+                int offset = Convert.ToInt32(context[_attributes["offset"]]);
                 collection = collection.Skip(offset);
             }
 
             if (_attributes.ContainsKey("limit"))
             {
-                int limit = Convert.ToInt32(_attributes["limit"]);
+                int limit = Convert.ToInt32(context[_attributes["limit"]]);
                 collection = collection.Take(limit);
             }
 
             collection = collection.ToList();
             int length = collection.Count();
 
-            int cols = Convert.ToInt32(context[_attributes["cols"]]);
+            int cols = _attributes.ContainsKey("cols") ? Convert.ToInt32(context[_attributes["cols"]]) : length;

```

Comment:
nit: While this is existing code, I recommend we update it to use full English words and avoid uncommon abbreviation such as this 'cols' for columns.

Reply:
What @daviburg is suggesting...
```suggestion
            int columns = _attributes.ContainsKey("cols") ? Convert.ToInt32(context[_attributes["cols"]]) : length;

```

Reply:
i changed the name